contentful/apps
# HAAs for mux

## Purpose

In order to move JWT token signing to a backend context, we want to set up the Mux app to use hosted app actions.

As a preliminary step, this PR focuses on reorganizing the app to include a front end along side the app actions, with a parent project that's responsible for the building and deployment of the whole thing.

## Approach

* Front end is moved to `frontend` folder
* HAA is created in `app-actions` folder
* Root contains a manifest pointing to a single "static" HAA (`muxGetSignedUrlTokens`)
* Action is implemented to return static values
* Copied and pasted a handful of boilerplate files etc. to operationalize the app action

Note: Signed URLs do not yet work in this version. Those will be fixed in a follow up.

## Testing steps

This PR was built and deployed to a sandbox space to verify that everything is still "correct":

<img width="821" alt="image" src="https://github.com/contentful/apps/assets/235836/326e2532-ef6a-41f2-89e7-dfcd9f6599a5">


A manual test of the app action in that space shows the action itself is deployed and works as expected.

```
ACCESS_TOKEN={redacted} npm run call-app-action -- -a muxGetSignedUrlTokens -s 0h02439dt6mi

> @contentful/mux-app@1.0.0 call-app-action
> npx ts-node scripts/call-app-action.ts -a muxGetSignedUrlTokens -s 0h02439dt6mi

APP ACTION RESULT
{
  sys: {
    id: '5bsgVGKFfgpPe7jnGj0DnA',
    type: 'AppActionCallDetails',
    space: { sys: { type: 'Link', linkType: 'Space', id: '0h02439dt6mi' } },
    sys: { type: 'Link', linkType: 'Environment', id: 'master' },
    createdBy: {
      sys: {
        type: 'Link',
        linkType: 'AppAction',
        id: 'muxGetSignedUrlTokens'
      }
    },
    createdAt: '2024-03-13T14:25:34.142Z'
  },
  request: { function: 'actions/get-jwt-tokens.js', parameters: {} },
  response: {
    body: '{"ok":true,"data":{"playbackToken":"token","posterToken":"token","storyboardToken":"token"}}',
    statusCode: 200
  },
  statusCode: 200,
  responseAt: '2024-03-13T14:25:34.756Z',
  errors: [],
  eventType: 'call',
  url: 'hosted',
  requestAt: '2024-03-13T14:25:34.142Z',
  environmentId: 'master'
}
Result: success!

```



## Breaking Changes



## Dependencies and/or References



## Deployment
